src/sources/util/wrappers.rs

```diff
@@ -0,0 +1,85 @@
+use pin_project::pin_project;
+use std::task::{Context, Poll};
+use std::{io, pin::Pin};
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+pub trait AfterReadExt {
+    fn after_read<F>(self, after_read: F) -> StreamWrapper<Self, F>
+    where
+        Self: Sized;
+}
+
+impl<T: AsyncRead + AsyncWrite> AfterReadExt for T {
+    fn after_read<F>(self, after_read: F) -> StreamWrapper<Self, F> {
```

Comment:
TIOLI: if we wanted to follow the common pattern for futures/streams combinators, the type would be called `AfterRead` instead of `StreamWrapper`.

Reply:
Good point, I'll change that.